code/datums/components/footstep.dm

```diff
@@ -107,20 +99,22 @@
 		return
 	var/mob/living/carbon/human/H = parent
 
+	var/step_type = null
+	var/list/step_sounds = null
```

Comment:
Something to make this ever so slightly more efficient, we can assign the else condition to `step_type` immediately.
```dm
var/step_type = H.dna.species.footstep_type

if((H.wear_suit?.body_parts_covered | H.w_uniform?.body_parts_covered | H.shoes?.body_parts_covered) & FEET)
	step_type = FOOTSTEP_MOB_SHOE // Feet are covered
```

Reply:
Same as above